prediction_market_agent_tooling/tools/contract.py

```diff
@@ -53,19 +57,37 @@ def wait_until_nonce_changed(
         time.sleep(sleep_time)
 
 
+T = TypeVar("T", bound="ContractBaseClass")
+
+
 class ContractBaseClass(BaseModel):
     """
     Base class holding the basic requirements and tools used for every contract.
     """
 
     CHAIN_ID: t.ClassVar[ChainID]
     CHAIN_RPC_URL: t.ClassVar[str]
-
-    abi: ABI
-    address: ChecksumAddress
+    signer_private_key: PrivateKey | None = (
+        None  # no private_key required for call methods (reading)
+    )
+    safe_address: ChecksumAddress | None = None  # no Safe address required, only signer_private_key suffices if not using Safe.
+    abi: ABI  # = Field(default=ABI(""))
```

Comment:
Why these new comments?

```suggestion
    abi: ABI
```